operator/v1/types_klusterlet.go

```diff
@@ -244,3 +244,64 @@ type KlusterletList struct {
 	// Items is a list of Klusterlet agents.
 	Items []Klusterlet `json:"items"`
 }
+
+const (
+	// The types of klusterlet condition status.
+	// ConditionKlusterletApplied is the klusterlet condition status which means all components have been applied on the managed cluster.
+	ConditionKlusterletApplied = "Applied"
+	// ConditionReadyToApply is a klusterlet condition status which means it is ready to apply the resources on the managed cluster.
+	ConditionReadyToApply = "ReadyToApply"
+	// ConditionKlusterletAvailable is the klusterlet condition status which means all components are available and ready to serve.
+	ConditionKlusterletAvailable = "Available"
+	// ConditionHubConnectionDegraded is the klusterlet condition status which means the agent on the managed cluster cannot access the hub cluster.
+	ConditionHubConnectionDegraded = "HubConnectionDegraded"
+	// ConditionRegistrationDesiredDegraded is the klusterlet condition status which means the registration agent on the managed cluster is not ready to server.
+	ConditionRegistrationDesiredDegraded = "RegistrationDesiredDegraded"
+	// ConditionWorkDesiredDegraded is the klusterlet condition status which means the work agent on the managed cluster is not ready to server.
+	ConditionWorkDesiredDegraded = "WorkDesiredDegraded"
+
+	// ReasonKlusterletApplied is the reason of ConditionKlusterletApplied condition to show resources are applied.
+	ReasonKlusterletApplied = "KlusterletApplied"
+	// ReasonKlusterletApplyFailed is the reason of ConditionKlusterletApplied condition to show failed to apply resources.
+	ReasonKlusterletApplyFailed = "KlusterletApplyFailed"
+	// ReasonKlusterletCRDApplyFailed is the reason of ConditionKlusterletApplied condition to show failed to apply CRDs.
+	ReasonKlusterletCRDApplyFailed = "CRDApplyFailed"
+	// ReasonManagedClusterResourceApplyFailed is the reason of ConditionKlusterletApplied condition to failed to apply resources on managed cluster.
+	ReasonManagedClusterResourceApplyFailed = "ManagedClusterResourceApplyFailed"
+	// ReasonManagementClusterResourceApplyFailed is the reason of ConditionKlusterletApplied condition  failed to apply resources on management cluster.
+	ReasonManagementClusterResourceApplyFailed = "ManagementClusterResourceApplyFailed"
+
+	// ReasonKlusterletPrepareFailed is the reason of ConditionReadyToApply condition to show failed to prepare before apply/delete resources.
```

Comment:
what does prepare mean?

Reply:
prepare means to get kubeconfig in external-managed-kubeconfig secret successfully 
changed to :
ReasonKlusterletPrepareFailed is the reason of ConditionReadyToApply condition to show it is failed to get the kubeConfig of managed cluster from the external-managed-kubeconfig secret in the hosted mode.

 ReasonKlusterletPrepared is the reason of ConditionReadyToApply condition to show the kubeConfig of managed cluster is validated from the external-managed-kubeconfig secret in the hosted mode.